Remove business plan upsell for premium monthly purchases

candy02058912 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/dotcom-forge/issues/2195

For the experiment, when purchasing monthly plans for personal/premium/business plans
* control: no upsell
* treatment: with annual upsell
* this has been discussed in pbxNRc-2tB-p2#comment-4248

## Proposed Changes

* Remove business plan upsell for premium monthly purchases

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Before


https://user-images.githubusercontent.com/6586048/234462616-9d26892d-e9e0-42d5-9f67-fa2a1d653f91.mp4



After


https://user-images.githubusercontent.com/6586048/234462346-439f8826-4264-4f88-814c-7ed877171aa8.mp4



* have a test account on a free/personal plan
* go to `/plans/monthly/:siteSlug`
* select the premium plan
* should see the after thank you page and not the upsell in before

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~
- [ ] ~Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~
- [ ] ~For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?~

candy02058912 committed:
Remove business plan upsell for premium monthly purchases

candy02058912 commented on client/my-sites/checkout/get-thank-you-page-url/test/get-thank-you-page-url.ts:
```diff
@@ -1013,7 +1013,10 @@ describe( 'getThankYouPageUrl', () => {
+	// NOTE: as part of the calypso_postpurchase_upsell_monthly_to_annual_plan experiment
+	// we're disabling the monthly premium to monthly business plan upsell
+	it( 'redirects to thank you page if jetpack is not in the cart, and premium monthly is in the cart', () => {
+		// it( 'redirects to business monthly upgrade nudge if jetpack is not in the cart, and premium monthly is in the cart', () => {
```
Unsure what's the best way to proceed with this, as this is for the `calypso_postpurchase_upsell_monthly_to_annual_plan` experiment. The current approach is depending on the experiment results, update the code here.

DustyReagan replied:
I kind of feel like we could work as if the Premium-Business monthly upsell removal is permanent (if that makes things easier to work with). Naturally, we can always change our mind after the experiment, and put it back, referring to these old PRs if needed.

DustyReagan reviewed (approved):
Tested and works as described! 👍

candy02058912 committed:
Add PR link to comment and remove commented code

candy02058912 commented:
I took the suggestion mentioned [here](https://github.com/Automattic/wp-calypso/pull/76260#discussion_r1177886215), and removed the commented-out code. Also added link to this PR.
